atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/scrub/KeyValueServiceScrubberStoreTest.java

```diff
@@ -0,0 +1,96 @@
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.cleaner.KeyValueServiceScrubberStore;
+import com.palantir.atlasdb.cleaner.ScrubberStore;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.impl.InMemoryKeyValueService;
+import com.palantir.common.base.BatchingVisitable;
+import com.palantir.common.base.BatchingVisitables;
+
+public class KeyValueServiceScrubberStoreTest {
+    private KeyValueService kvs;
+    private ScrubberStore scrubStore;
+
+    @Before
+    public void before() {
+        kvs = new InMemoryKeyValueService(false, MoreExecutors.newDirectExecutorService());
+        scrubStore = KeyValueServiceScrubberStore.create(kvs);
+    }
+
+    @After
+    public void after() {
+        kvs.close();
+    }
+
+    @Test
+    public void testEmptyStoreReturnsNothing() {
+        Assert.assertEquals(ImmutableList.of(), getScrubQueue());
+    }
+
+    @Test
+    public void testSingleEntryStore() {
+        Cell cell = Cell.create(new byte[] {1, 2, 3}, new byte[] {4, 5, 6});
+        TableReference ref = TableReference.fromString("foo.bar");
+        long timestamp = 10;
+        scrubStore.queueCellsForScrubbing(ImmutableMultimap.of(cell, ref), timestamp, 1000);
+        Assert.assertEquals(
+                ImmutableList.of(ImmutableSortedMap.of(timestamp, ImmutableMultimap.of(ref, cell))),
+                getScrubQueue());
+        scrubStore.markCellsAsScrubbed(ImmutableMap.of(ref, ImmutableMultimap.of(cell, timestamp)), 1000);
+        Assert.assertEquals(ImmutableList.of(), getScrubQueue());
+    }
+
+    @Test
+    public void testMultipleEntryStore() {
+        Cell cell1 = Cell.create(new byte[] {1, 2, 3}, new byte[] {4, 5, 6});
+        Cell cell2 = Cell.create(new byte[] {7, 8, 9}, new byte[] {4, 5, 6});
+        Cell cell3 = Cell.create(new byte[] {1, 2, 3}, new byte[] {7, 8, 9});
+        TableReference ref1 = TableReference.fromString("foo.bar");
+        TableReference ref2 = TableReference.fromString("foo.baz");
+        long timestamp1 = 10;
+        long timestamp2 = 20;
+        scrubStore.queueCellsForScrubbing(ImmutableMultimap.of(cell1, ref1, cell2, ref1), timestamp1, 1000);
+        scrubStore.queueCellsForScrubbing(ImmutableMultimap.of(cell1, ref1, cell3, ref2), timestamp2, 1000);
+        Assert.assertEquals(
+                ImmutableList.of(ImmutableSortedMap.of(
+                        timestamp1, ImmutableMultimap.of(ref1, cell2),
+                        timestamp2, ImmutableMultimap.of(ref2, cell3, ref1, cell1))),
+                getScrubQueue());
+        scrubStore.markCellsAsScrubbed(ImmutableMap.of(
+                ref2, ImmutableMultimap.of(cell3, timestamp2),
+                ref1, ImmutableMultimap.of(cell1, timestamp1)), 1000);
```

Comment:
Sure, the test is just checking that after deleting some stuff the scrub queue is in the expected state. I was not carefully making sure to delete the same stuff the scrubber would. I'll do that though.